runtime/js-compute-runtime/builtins/backend.cpp

```diff
@@ -806,41 +806,153 @@ JSString *Backend::name(JSContext *cx, JSObject *self) {
 }
 
 bool Backend::toString(JSContext *cx, unsigned argc, JS::Value *vp) {
-  METHOD_HEADER(0)
+  METHOD_HEADER(0);
   JS::RootedString name(cx, JS::GetReservedSlot(self, Backend::Slots::Name).toString());
   args.rval().setString(name);
   return true;
 }
 
-const JSFunctionSpec Backend::static_methods[] = {
-    JS_FS_END,
-};
-
-const JSPropertySpec Backend::static_properties[] = {
-    JS_PS_END,
-};
-
-const JSFunctionSpec Backend::methods[] = {JS_FN("toString", toString, 0, JSPROP_ENUMERATE),
-                                           JS_FS_END};
-
-const JSPropertySpec Backend::properties[] = {JS_PS_END};
-
-bool Backend::set_name(JSContext *cx, JSObject *backend, JS::HandleValue name_val) {
+namespace {
+JSString *parse_and_validate_name(JSContext *cx, JS::HandleValue name_val) {
```

Comment:
Do you know if returning a `JSString *` is safe here? Its value is guaranteed to not move while the `JS::RootedString` is present, but once it's returned I'm not sure there's anything that would prevent the GC from moving the value.

Perhaps and easier change would be to return a `host_api::HostString`, given that each use of this function immediately calls `core::encode` on the return value?